Pin the version of meson

dsdolzhenko opened the pull request:
The PR fixates the version of Meson to prevent unexpected built failures like [this one](https://github.com/versatica/mediasoup/pull/857) in future.

dsdolzhenko committed:
Pin the version of meson

nazar-pc reviewed (approved):
Thank you!

dsdolzhenko commented:
@nazar-pc Do you think we need to pin the versions of `ninja`, `setuptools` and `pip` as well?

nazar-pc commented:
Strange CI error :thinking:

nazar-pc commented:
> @nazar-pc Do you think we need to pin the versions of ninja, setuptools and pip as well?

I don't think so

ibc commented:
Should I publish a new mediasoup release even before this PR lands?

nazar-pc commented:
I think we might need to downgrade Meson, looks like latest version doesn't support Ubuntu 18.04 for some reason unless we are ready to drop support for 18.04, which I don't think we're ready to do just yet.

dsdolzhenko commented:
> I think we might need to downgrade Meson, looks like latest version doesn't support Ubuntu 18.04 for some reason unless we are ready to drop support for 18.04, which I don't think we're ready to do just yet.

Starting from [0.62.0rc1](https://pypi.org/project/meson/0.62.0rc1/#history) Meson requires Python 3.7. So, yes we can try to  downgrade to `0.61.5`

nazar-pc commented:
Yes, let's try that for now, we can bump requirements later, but we need to fix this for current users ASAP

dsdolzhenko committed:
Downgrade the version of Meson to 0.61.5

That is the latest version of Meson supporting Python 2.6 which is default version on Ubuntu 18.04

ibc commented:
Will merge and release ASAP

eli-schwartz commented on worker/Makefile:
```diff
@@ -74,7 +75,7 @@ ifeq ($(wildcard $(PIP_DIR)),)
 	# Install `meson` and `ninja` using `pip` into custom location, so we don't
 	# depend on system-wide installation.
-	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) $(PIP_BUILD_BINARIES) meson ninja
+	$(PYTHON) -m pip install --upgrade --target=$(PIP_DIR) $(PIP_BUILD_BINARIES) meson==$(MESON_VERSION) ninja
```
Locking to an exact version means that pip cannot cleverly figure out which ones are compatible with your current system. A far, far, far better idea:

```
pip install "meson>=0.58,<=0.62.*"
```

This pins an allowable minimum/maximum range, which supports reasonably new Meson on systems with modern Python, while also handling archaic systems such as Ubuntu 18.04.

That being said, I'm probably duty-bound to say I'm not a fan of pinning versions... you'll need to babysit this to make sure it gets upgraded regularly, and I don't think tools such as dependabot or renovatebot (https://mend.io) can detect version pins as arbitrary lines in a Makefile.

ibc commented:
3.10.1 released

nazar-pc commented:
Rust version 0.10.0 released too
